pkg/providers/anthropic/client.go

```diff
@@ -40,6 +41,7 @@ func NewClient(providerConfig *Config, clientConfig *clients.ClientConfig, tel *
 		chatURL:             chatURL,
 		config:              providerConfig,
 		chatRequestTemplate: NewChatRequestFromConfig(providerConfig),
+		errMapper:           NewErrorMapper(tel),
```

Comment:
I like this

Reply:
It was time to break this thing into a smaller component. We may even end up having one error mapper if there is no much difference between error mapping across providers 👀